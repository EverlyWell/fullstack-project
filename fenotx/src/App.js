import React, { Component } from 'react';
import axios from 'axios';
import './App.css';

axios.defaults.baseURL = "http://localhost:3020/api/v1/cats"

class App extends Component {

  async getBreeds() {
    const res = await axios('/breeds');
    return res.data;
  }
  async getCategories() {
    const res = await axios('/categories');
    return res.data;
  }
  async getCatsImagesByBreed(breed_id, amount) {
    const res = await axios('/search_by_breed', { params: { breed_id: breed_id, limit: amount }});
    return res.data;
  }
  async getCatsImages(breed_id, category_id, amount) {
    const res = await axios('/search', {
      params: { breed_id: breed_id, category_id: category_id, limit: amount }
    });
    return res.data;
  }

  async loadBreedImages() {
    let breed_images = '';
    if(this.state.selected_category === 'none'){
      breed_images = await this.getCatsImagesByBreed(this.state.selected_breed, 5);
    }else{
      breed_images = await this.getCatsImages(this.state.selected_breed, this.state.selected_category, 5);
    }
    this.setState({ images: breed_images });
  }

  constructor(...args) {

    super(...args);
    this.state = {
        images: [],
        breeds: [],
        categories: [],
        selected_breed: 'abys',
        selected_category: 'none',
    };

    this.onBreedSelectChange = this.onBreedSelectChange.bind(this);
    this.onCategorySelectChange = this.onCategorySelectChange.bind(this);
  }
  async onBreedSelectChange(e) {
    await this.setState({selected_breed: e.target.value});
    await this.loadBreedImages();
  }
  async onCategorySelectChange(e) {
    await this.setState({selected_category: e.target.value});
    await this.loadBreedImages();
  }
  componentDidMount() {
    if (this.state.breeds.length===0) {
      (async () => {
        try {
          this.setState({ breeds: await this.getBreeds() });
          this.setState({ categories: await this.getCategories() });
        } catch (e) {
          //...handle the error...
          console.error(e)
        }
      })();
    }
  }
  render() {
    return (
      <div>
        Breed:
        <select
          value={this.state.selected_breed}
          onChange={this.onBreedSelectChange}
        >
          { this.state.breeds.map((breed) =>
            <option key={breed.id} value={breed.id}>{breed.name}</option>) }
        </select>

        &nbsp;&nbsp;

        Category:
        <select
          value={this.state.selected_category}
          onChange={this.onCategorySelectChange}
        >
          <option value='none'>None</option>
          { this.state.categories.map((category) =>
            <option key={category.id} value={category.id}>{category.name}</option>) }
        </select>

        <div className="images">
          { this.state.images.map((image, i) => (
            <img key={i} className="cat-image" alt="" src={image.url}></img>
          ))}
        </div>

      </div>
    );
  }
}

export default App;
