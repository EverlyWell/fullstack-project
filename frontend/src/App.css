* {
  box-sizing: border-box;
  background-color: rgb(244, 244, 244);
  color: #333;
  font-size: 10px;
}

.App {
  margin: 0;
  padding: 0;
}

.container {
  margin: 0 auto;
  max-width: 1000px;
  padding: 40px;
}

.title {
  font-size: 4.4rem;
  font-family: "Gill Sans", "Gill Sans MT", Calibri, "Trebuchet MS", sans-serif;
}

.form {
  display: grid;
}

.label {
  font-size: 3rem;
  margin-bottom: 1rem;
}

.input {
  font-size: 1.6rem;
  padding: 0.5rem 2rem;
  line-height: 2.8rem;
  border-radius: 20px;
  background-color: white;
  margin-bottom: 1rem;
}

.button {
  background-color: rgba(0, 0, 0, 0.75);
  color: white;
  padding: 1rem 2rem;
  border: 1px solid rgba(0, 0, 0, 0.75);
  border-radius: 20px;
  font-size: 1.4rem;
  cursor: pointer;
  transition: background-color 250ms;
}

.button:hover {
  background-color: rgba(0, 0, 0, 0.85);
}

.card-list {
  column-count: 3;
}

.card {
  margin-bottom: 1rem;
  display: flex;
}

.card--image {
  flex: 100%;
  margin-top: 1rem;
  border-radius: 10px;
}

@media (min-width: 768px) {
  .form {
    grid-template-columns: auto 1fr auto;
    grid-gap: 1rem;
    align-items: center;
  }
  .input {
    margin-bottom: 0;
  }
}

@media only screen and (max-width: 600px) {
    .card-list {
        column-count: 1;
    }
}