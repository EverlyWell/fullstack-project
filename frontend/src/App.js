import React, {useState,useEffect} from 'react';
import './App.css';
import axios from 'axios'

import SearchBar from "./components/SearchBar"
import GifsList from "./components/GifsList"
import Favorites from "./components/Favorites" 

function App() {
  const [gifs, setGifs] = useState([])
  const [searchTerm, setSearchTerm] = useState('')

  return (
    <div className="App">
      <SearchBar />
      <GifsList />
      <Favorites />
    </div>
  );
}

export default App;
