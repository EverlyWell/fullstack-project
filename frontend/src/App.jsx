import React from 'react'
import './App.css'

// Redux
import { connect } from 'react-redux'
import { setAuthToken, setAuthLoaded } from './actions/index'

// Libraries
import Cookies from 'js-cookie'

// Library components
import {
  BrowserRouter as Router,
  Switch
} from "react-router-dom"
import Container from '@material-ui/core/Container'

// Child components
import Login from './Login'
import Home from './Home'
import Favorites from './Favorites'
import AuthenticatedRoute from './AuthenticatedRoute'
import UnauthenticatedRoute from './UnauthenticatedRoute'

function App({ setAuthToken, setAuthLoaded }) {
  let authToken = Cookies.get('userToken')
  if(authToken && authToken.length) {
    setAuthToken(authToken)
  }
  setAuthLoaded(true)

  return (
    <Router>
      <Container component="main">
        <Switch>
          <UnauthenticatedRoute path="/login">
            <Login screen="login" />
          </UnauthenticatedRoute>

          <UnauthenticatedRoute path="/signup">
            <Login screen="signup" />
          </UnauthenticatedRoute>

          <AuthenticatedRoute path="/favorites">
            <Favorites />
          </AuthenticatedRoute>

          <AuthenticatedRoute path="/">
            <Home />
          </AuthenticatedRoute>
        </Switch>
      </Container>
    </Router>
  )
}

export default connect(
  null,
  { setAuthToken, setAuthLoaded }
)(App)
