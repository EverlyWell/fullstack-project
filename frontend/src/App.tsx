import React, {useState, useEffect, useRef} from 'react';
import './App.css';
import ImagesList from './components/ImagesList';
import axios from "axios";
import Cookies from 'js-cookie';
import Login from "./components/LogIn";
import ImageFilters from "./components/ImageFilters";
import {Filters} from './types'

function App() {
    const [images, setImages] = useState([]);
    const [loggedIn, setLoggedIn] = useState(false);

    useEffect(()=>{
        const jwt = Cookies.get('jwt');
        if(jwt!==undefined){
            setLoggedIn(true);
            loadImages();
        } else {
            console.log('not logged in');
        }
    }, [])

    const loadImages = async (url: string = 'api/images', filters: Filters = {}) => {
       const res = await axios.get(url,
           {params: filters, headers: {'Authorization': `Bearer ${Cookies.get('jwt')}`}
       });
       setImages(res.data);
    }

    return (
        <div className="App">
            <div className="LoginContainer">
                <Login loggedIn={loggedIn} setLoggedIn={setLoggedIn}/>
             </div>
            { loggedIn &&
                <div>
                    <ImageFilters loadImages={loadImages}/>
                    <hr/>
                    <ImagesList images={images}/>
                </div>
            }
        </div>
    );
}

export default App;