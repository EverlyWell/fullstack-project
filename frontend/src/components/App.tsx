import React, { useState, useEffect } from 'react';
import { favorite, unFavorite } from '../api';
import Search from './Search';
import Favorites from './Favorites';
import '../App.css';

function App() {
  const [page, setPage] = useState('search');
  const [favorites, setFavorites] = useState([] as any);
  
  useEffect(() => {
    fetch('/api/favorites')
      .then(r => r.json())
      .then(({ favorites: favs }) => {
        setFavorites(favs.map((f: any) => f.giphy_id))
      });
  }, []);
  
  const isGifFav = (id: string) => favorites.includes(id);

  const toggleFavorite = (id: string) => {
    if (isGifFav(id)) {
      unFavorite(id)
      .then(() => setFavorites(favorites.filter((e: string) => e !== id)))
      .catch(e => console.error(e));
    } else {
      favorite(id)
      .then(() => setFavorites([...favorites, id]))
      .catch(e => console.error(e));
    }
  }
  
  const className = (id: string) => page === id ? 'selected' : '';

  return (
    <div className="App">
    <ul id="nav">
      <li className={className('search')}><a href="#search" onClick={() => setPage('search')}>Search</a></li>
      <li className={className('fav')}><a href="#favorites" onClick={() => setPage('fav')}>Favorites</a></li>
    </ul>
    {page === 'search' ?
      <Search isGifFav={isGifFav} toggleFavorite={toggleFavorite} /> :
      <Favorites collection={favorites} isGifFav={isGifFav} toggleFavorite={toggleFavorite} />
    }
    </div>
  )
}

export default App;
