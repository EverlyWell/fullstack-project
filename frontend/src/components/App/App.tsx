import { FunctionComponent, useState } from 'react';
import './App.css';
import Container from '@mui/material/Container';
import Typography from '@mui/material/Typography';
import { Box, Paper, Tab, Tabs,  } from '@mui/material';
import { TabContext, TabList, TabPanel  } from '@mui/lab';
import { ICategory } from '../../typings/typings';
import FiltersView from '../Filters/FiltersView';
import CatListView from '../CatsList/CatsListView';
import FavsListView from '../FavsList/FavsListView';

interface IApp {
}

const App: FunctionComponent<IApp> = () => {
  const [category, setCategory] = useState<ICategory>();
  const [tabValue, setTabValue] = useState<string>('cats');

  const handleCategorySelect = (selectedCategory: ICategory) => setCategory(selectedCategory);
  const handleTabValueChange = (event: any, newValue: string) => setTabValue(newValue);

  return (
    <Container maxWidth="md" className="App">
      <Paper>
        <Typography variant="h4" component="h1" gutterBottom>
          The Cats API challenge
        </Typography>
        <FiltersView
          handleCategorySelect={handleCategorySelect}
        />
        <Box sx={{ width: '100%' }}>
          <TabContext value={tabValue}>
            <TabList onChange={handleTabValueChange} aria-label="lab API tabs example">
              <Tab label="Cats" value="cats" />
              <Tab label="Favourites" value="favourites" />
            </TabList>
            <TabPanel value="cats">
              <CatListView
                category={category}
              />
            </TabPanel>
            <TabPanel value="favourites">
              <FavsListView
                category={category}
              />
            </TabPanel>
          </TabContext>
        </Box>
      </Paper>
    </Container>
  );
}

export default App;
