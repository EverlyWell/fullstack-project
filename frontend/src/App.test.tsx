import React from 'react';
import { render } from '@testing-library/react';
import App from './App';

test('renders search form', () => {
  const { getByText } = render(<App />);
  const searchElement = getByText(/search/i);
  expect(searchElement).toBeInTheDocument();
});
