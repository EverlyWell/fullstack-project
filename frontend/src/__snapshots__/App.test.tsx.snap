// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders successfully 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <h1>
      Cute collector
    </h1>
  </header>
</div>
`;
